custom_components/solcast_solar/coordinator.py

```diff
@@ -47,6 +48,8 @@ async def setup(self):
             #4.0.18 - added reset usage call to reset usage sensors at UTC midnight
             async_track_utc_time_change(self._hass, self.update_utcmidnight_usage_sensor_data, hour=0,minute=0,second=0)
             async_track_utc_time_change(self._hass, self.update_integration_listeners, minute=range(0, 60, 5), second=0)
+            #4.0.41 - recalculate splines at midnight local
+            async_track_time_change(self._hass, self.update_midnight_spline_recalc, hour=0,minute=0,second=0)
```

Comment:
There was already a method executed at midnight, we could have put the code in there instead of adding another trigger.

Reply:
UTC vs. locaal time...